x-pack/plugins/security_solution/public/management/pages/host_isolation_exceptions/service.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  ExceptionListItemSchema,
+  FoundExceptionListItemSchema,
+} from '@kbn/securitysolution-io-ts-list-types';
+import { ENDPOINT_HOST_ISOLATION_EXCEPTIONS_LIST_ID } from '@kbn/securitysolution-list-constants';
+import { HttpStart } from 'kibana/public';
+import { EXCEPTION_LIST_ITEM_URL, EXCEPTION_LIST_URL } from '../event_filters/constants';
+import { HOST_ISOLATION_EXCEPTIONS_LIST } from './constants';
+
+let listExists = false;
+
+async function createHostIsolationExceptionList(http: HttpStart) {
+  try {
+    await http.post<ExceptionListItemSchema>(EXCEPTION_LIST_URL, {
+      body: JSON.stringify(HOST_ISOLATION_EXCEPTIONS_LIST),
+    });
+  } catch (err) {
+    // Ignore 409 errors. List already created
+    if (err.response?.status !== 409) {
+      throw err;
+    }
+  }
+  listExists = true;
+}
+
+async function ensureHostIsolationExceptionsListExists(http: HttpStart) {
+  if (listExists) {
+    return;
+  }
+  return createHostIsolationExceptionList(http);
+}
```

Comment:
It is, but this will give an eslint-error that's why I didn't go for it.

Reply:
So I tried this myself it doesn't give me any errors. What TS error do you see?

Reply:
It is not a TS error. It is a eslint error.

But based on your advice I modified this function to now use a cached promise instead of a true/false flag.